src/lib/storage/index/abstract_table_index.hpp

```diff
@@ -79,68 +76,82 @@ class AbstractTableIndex : private Noncopyable {
 
   AbstractTableIndex() = delete;
   explicit AbstractTableIndex(const TableIndexType type);
-  AbstractTableIndex(AbstractTableIndex&&) = default;
+  AbstractTableIndex(AbstractTableIndex&&) = delete;
   virtual ~AbstractTableIndex() = default;
 
   /**
-   * Searches for all positions of the entry within the table index.
-   *
-   * Calls _equals() of the derived class.
-   * @param value used to query the index.
-   * @return A pair of Iterators containing the start and end iterator for the stored RowIDs of the element inside the
-   * table index.
+   * The following four methods are used to access any table index. Each of them accepts a generic function object (or
```

Comment:
What is something similar to a function object? A function object is already quite general:
`A function object is any object for which the function call operator is defined.` https://en.cppreference.com/w/cpp/utility/functional